src/statusbar.cpp

```diff
@@ -56,3 +56,10 @@ void StatusBar::setTheme(Theme theme)
     StatusBarPrivate::theme = theme;
     StatusBarPrivate::setTheme_sys(theme);
 }
+
+void registerQmlStatusBar()
```

Comment:
i left this out to avoid a heavy QtQml dependency for users that would not use QML, but we could provide it as an inline function guarded with QT_QML_LIB to make the dependency optional. furthermore, it would be also nice to let users also control the namespace:

statusbar.h:

    #ifdef QT_QML_LIB
    #include <QtQml/qqml.h>
    #endif

    class StatusBar : public QObject
    {
        // ...
    
    #ifdef QT_QML_LIB
        static void registerQmlType(const char *uri = "StatusBar", int majorVersion = 1, int minorVersion = 0)
        {
            qmlRegisterType<StatusBar>(uri, majorVersion, minorVersion, "StatusBar");
        }
    #endif
    };

then it's a matter of:

    #include "statusbar.h"

    // ...

    StatusBar::registerQmlType();
    // or:
    // StatusBar::registerQmlType("MyOwnNamespace");
    // or:
    // StatusBar::registerQmlType("MyOwnNamespace", 2, 0);

what do you think?

Reply:
Sorry, I didn't know about `QT_QML_LIB` define, but I still think that its right way to register qml type automatically just by adding lib to build system. I also think that its bad practice to register external library class in own namespace.
So I prefer to hide all qml related stuff under `#ifdef QT_QML_LIB`, but left registration to `Q_COREAPP_STARTUP_FUNCTION`.
What do you think?

Reply:
The "right way" would be to build a QML module that does the registration. Anyway, what about leaving this to a separate PR? The CMake-related changes could be merged...